packages/pg-v5/lib/bastion.js

```diff
@@ -81,37 +83,54 @@ function getConfigs (db) {
 
 exports.getConfigs = getConfigs
 
-function sshTunnel (db, dbTunnelConfig, timeout) {
-  return new Promise((resolve, reject) => {
-    // if necessary to tunnel, setup a tunnel
-    // see also https://github.com/heroku/heroku/blob/master/lib/heroku/helpers/heroku_postgresql.rb#L53-L80
-    let timer = setTimeout(() => reject(new Error('Establishing a secure tunnel timed out')), timeout)
-    if (db.bastionKey) {
-      let tun = tunnel(dbTunnelConfig, (err, tnl) => {
-        if (err) {
-          debug(err)
-          reject(new Error('Unable to establish a secure tunnel to your database.'))
-        }
-        debug('Tunnel created')
-        clearTimeout(timer)
-        resolve(tnl)
-      })
-      tun.on('error', (err) => {
-        // we can't reject the promise here because we may already have resolved it
-        debug(err)
-        cli.exit(1, 'Secure tunnel to your database failed')
-      })
-    } else {
-      clearTimeout(timer)
-      resolve()
+class Timeout {
+  constructor (timeout, message) {
+    this.timeout = timeout
+    this.message = message;
+    this.events = new EventEmitter()
+  }
+
+  async promise () {
+    this.timer = setTimeout(() => {
+      this.events.emit('error', new Error(this.message))
+    }, this.timeout)
+
+    try {
+      await once(this.events, 'cancelled')
```

Comment:
[events.once](https://nodejs.org/docs/latest-v14.x/api/events.html#events_events_once_emitter_name) will return a promise and resolve with the value of the event. If, while it is waiting for the event, an `error` event occurs, it will reject the promise. That's why you don't see the `.on('error')` event handler carried over in the new version.